cdap-app-fabric/src/main/java/io/cdap/cdap/app/store/Store.java

```diff
@@ -473,6 +473,13 @@ boolean scanApplications(ScanApplicationsRequest request, int txBatchSize,
    */
   void removeApplication(ApplicationId id);
 
+  /**
+   * Removes the latest version record of an application.
+   *
+   * @param appRef Application reference
+   */
+  void removeLatestAppRecord(ApplicationReference appRef);
```

Comment:
Because we have the FF in `DELETE "/apps/{app-id}/versions/{version-id}")`, we still allow users to delete a certain version of an app, we cannot simply change `removeAppliaction(ApplicationId)` to `removeApplication(ApplicationReference)`. And I think the current delete handler already has issues, especially the DELETE namespace one, we need to fix it in the while we asyncly delete all app versions.